onyo/lib/onyo.py

```diff
@@ -906,6 +930,227 @@ def valid_name(self, asset: Path) -> bool:
+
+        if error_path_protected:
+            log.error('The following paths are protected by onyo:\n' +
+                      '\n'.join(error_path_protected) + '\n' +
+                      'No directories were created.')
+            raise OnyoProtectedPathError('The following paths are protected by onyo:\n' +
+                                         '\n'.join(error_path_protected))
+
+        if len(paths_to_set) == 0:
+            log.error("No assets selected.")
+            raise ValueError("No assets selected.")
+
+        return paths_to_set
+
+    def _update_names(self, assets: list[Path], name_values: dict) -> None:
+        """
+        Set the pseudo key fields of an assets name (rename an asset file) from
+        values of a dictionary and test that the new name is valid and
+        available.
+        """
+        new_assets = []
+
+        # count and request the needed faux serial numbers
+        faux_serial_list = []
+        if 'serial' in name_values.keys() and name_values['serial'] == 'faux':
+            faux_number = len(assets)
+            if faux_number > 0:
+                faux_serial_list = self.generate_faux_serials(num=faux_number)
+
+        for asset in assets:
+            # split old name into parts
+            [serial, model, make, type] = [field[::-1] for field in re.findall('(.*)\.(.*)_(.*)_(.*)', asset.name[::-1])[0]]
+            fields = name_values.keys()
+
+            # update name fields and build new asset name
+            if "serial" in fields:
+                if name_values["serial"] == "faux":
+                    serial = faux_serial_list.pop()
+                else:
+                    serial = name_values["serial"]
+            if "model" in fields:
+                model = name_values["model"]
+            if "make" in fields:
+                make = name_values["make"]
+            if "type" in fields:
+                type = name_values["type"]
+            new_name = Path(asset.parent, f"{type}_{make}_{model}.{serial}")
+
+            # Check validity of the new asset name
+            if new_name == asset.name:
+                log.error(f"New asset names must be different than old names: '{new_name}'")
+                raise ValueError(f"New asset names must be different than old names: '{new_name}'")
+
+            if not self.valid_name(new_name):
+                log.error(f"New asset name is not valid: '{new_name}'")
+                raise ValueError(f"New asset name is not valid: '{new_name}'")
+
+            self.valid_asset_path_and_name_available(new_name, new_assets)
+            new_assets.append(new_name)
+
+            self._git(["mv", str(asset), str(new_name)])
+
+    def _write_asset(self, asset: Path, contents: dict) -> None:
+        """
+        Write contents into an asset file.
+        """
+        if contents == {}:
+            asset.open('w').write("")
+        else:
+            yaml = YAML(typ='rt')
```

Comment:
Does round-trip mode work here? The object isn't shared across `_read_asset()` and `_write_asset()`...